modules/repository/repo.go

```diff
@@ -71,7 +71,15 @@ func MigrateRepositoryGitData(ctx context.Context, u *user_model.User,
 		return repo, fmt.Errorf("Failed to remove %s: %v", repoPath, err)
 	}
 
-	if err = git.CloneWithContext(ctx, opts.CloneAddr, repoPath, git.CloneRepoOptions{
+	cargs := make([]string, len(git.GlobalCommandArgs))
+	copy(cargs, git.GlobalCommandArgs)
+
+	if opts.UseGitCredentials {
+		// TODO: replace `echo`` with something more generic that could be used on windows.
+		cargs = append(cargs, "-c", fmt.Sprintf("credential.helper='echo -e \"username=%s\\npassword=%s\\n\"'", opts.AuthUsername, opts.AuthPassword))
```

Comment:
Should look like:
"credential.helper=/bin/echo -e \"username=%s\\npassword=%s\\n\""

- Full path is required or git interprets them differently (man gitcredentials). Using ! is not option either as git command is not running under bash...
- Single quotes should be removed because they becomes part of argument

man 7 gitcredentials:
```
The name of an external credential helper, and any associated options. If the helper name is not an absolute path, then the string git credential- is prepended. The resulting string is executed by the shell (so, for example, setting this to foo --option=bar will execute git credential-foo --option=bar via the shell. See the manual of specific helpers for examples of their use.
```